ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/KeepAlive/Direct.hs

```diff
@@ -0,0 +1,19 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Ouroboros.Network.Protocol.KeepAlive.Direct where
+
+import Ouroboros.Network.Protocol.KeepAlive.Client
+import Ouroboros.Network.Protocol.KeepAlive.Server
+
+direct :: Monad m
```

Comment:
More for my edification, why did you choose to call this function `direct`?

Reply:
Because it directly connects the client ADT and server ADT without passing through Peer (where a similliar function is called `connect`, so the name is taken), or using any kind of channel.  This is a convention shared between all mini-protocol tests.